### PR TITLE
Add additional commands for Frequent Income feature

### LINKED ISSUE #158: Delete frequent incomes
As a user, I want to be able to delete frequent incomes so that I can remove frequent incomes that I no longer want to keep.

### PR DESCRIPTION
**Pull request by siddarth2824**
Resolves #158, #159 and #160.

Changes:
- Added `EditFrequentIncomeCommand` for users to edit existing frequent incomes
- Added `DeleteFrequentIncomeCommand` for users to delete existing frequent incomes
- Added `ConvertFrequentIncomeCommand` for users to convert the specified frequent income and add it to the finance trackers as an income
- Added parsers for the above mentioned commands to validate user inputs
- Added tests for:
    - `EditFrequentIncomeCommand`
    - `DeleteFrequentIncomeCommand`
    - `ConvertFrequentIncomeCommand`
    - `EditFrequentIncomrCommandParser`
    - `DeletetFrequentIncomrCommandParser`
    - `ConvertFrequentIncomrCommandParser`

Future Work:
- Restrict users from utilising `EditFrequentIncomeCommand`, `DeleteFrequentIncomeCommand` and `ConvertFrequentIncomeCommand`only under the `Income` tab

### CODE REVIEW

**Review by wltan — DISMISSED** [review]
I don't think it is good from a UX perspective that the user is able to convert incomes (and expenses also) from a list that is not even visible to them.

What I envision is that these commands should be restricted to their respective tabs (as is the case for `edit` and `delete` currently). Then, we can add a generic command that is simpler to use (also the case for `edit` and `delete` currently).

But that can be addressed in a future issue. The logic behind the commands LGTM!

Edit: You may want to check if #158, #159, #160 should be tagged `type.story` instead.

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequent/DeleteFrequentIncomeCommand.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+
+import ay2021s1_cs2103_w16_3.finesse.commons.core.index.Index;
+import ay2021s1_cs2103_w16_3.finesse.logic.commands.Command;
+import ay2021s1_cs2103_w16_3.finesse.logic.commands.CommandResult;
+import ay2021s1_cs2103_w16_3.finesse.logic.commands.exceptions.CommandException;
+import ay2021s1_cs2103_w16_3.finesse.model.Model;
+import ay2021s1_cs2103_w16_3.finesse.model.frequent.FrequentIncome;
+
+public class DeleteFrequentIncomeCommand extends Command {
```
Comment (yongping827) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequent/DeleteFrequentIncomeCommand.java:15]: Missing Javadocs

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/frequent/FrequentIncomeList.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import java.util.List;
```
Comment (yongping827) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/frequent/FrequentIncomeList.java:7]: Not all methods from `FrequentExpenseList` are mirrored here.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentExpenseCommandTest.java [hunk 3]
```diff
@@ -86,8 +86,8 @@ public void execute_invalidFrequentExpenseIndexUnfilteredList_failure() {
 
 
     /**
-     * Edit filtered list where index is larger than size of filtered list,
-     * but smaller than size of address book
+     * Edit filtered expense list where index is larger than size of filtered expense list,
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentExpenseCommandTest.java:89]:
```suggestion
     * Edit filtered frequent expense list where index is larger than size of filtered frequent expense list,
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentIncomeCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        EditFrequentTransactionDescriptor descriptor = new EditFrequentTransactionDescriptorBuilder()
+                .withTitle(VALID_TITLE_PART_TIME).build();
+        EditFrequentIncomeCommand editFrequentIncomeCommand =
+                new EditFrequentIncomeCommand(outOfBoundIndex, descriptor);
+
+        assertCommandFailure(editFrequentIncomeCommand, model, MESSAGE_INVALID_FREQUENT_INCOME_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered income list where index is larger than size of filtered income list,
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentIncomeCommandTest.java:89]:
```suggestion
     * Edit filtered frequent income list where index is larger than size of filtered frequent income list,
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentTransactionDescriptorTest.java [hunk 5]
```diff
@@ -34,17 +35,18 @@ public void equals() {
         assertFalse(DESC_PHONE_BILL.equals(DESC_SPOTIFY_SUBSCRIPTION));
 
         // different title -> returns false
-        EditFrequentExpenseDescriptor editedFrequentExpense = new EditFrequentExpenseDescriptorBuilder(DESC_PHONE_BILL)
+        EditFrequentTransactionDescriptor editedFrequentExpense =
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentTransactionDescriptorTest.java:38]: Use a more descriptive variable name.

**Review by yongping827 — CHANGES_REQUESTED** [review]

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentTransactionDescriptorTest.java [hunk 6]
```diff
@@ -34,18 +35,19 @@ public void equals() {
         assertFalse(DESC_PHONE_BILL.equals(DESC_SPOTIFY_SUBSCRIPTION));
 
         // different title -> returns false
-        EditFrequentExpenseDescriptor editedFrequentExpense = new EditFrequentExpenseDescriptorBuilder(DESC_PHONE_BILL)
+        EditFrequentTransactionDescriptor editedFrequentTransaction =
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/frequentcommands/EditFrequentTransactionDescriptorTest.java:38]:
```suggestion
        EditFrequentTransactionDescriptor editedFrequentTransactionDescriptor =
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/model/frequent/FrequentIncomeListTest.java [hunk 7]
```diff
@@ -14,6 +14,7 @@
 import org.junit.jupiter.api.Test;
 
 import ay2021s1_cs2103_w16_3.finesse.model.frequent.exceptions.DuplicateFrequentTransactionException;
+import ay2021s1_cs2103_w16_3.finesse.model.frequent.exceptions.FrequentTransactionNotFoundException;
 import ay2021s1_cs2103_w16_3.finesse.testutil.FrequentTransactionBuilder;
 
 public class FrequentIncomeListTest {
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/model/frequent/FrequentIncomeListTest.java:20]: Not all tests in `FrequentExpenseListTest` are mirrored here.

**Review by yongping827 — CHANGES_REQUESTED** [review]

**Review by yongping827 — APPROVED** [review]

**Review by wltan — APPROVED** [review]
LGTM

**Pull request merged**